src/crepe/api/Button.h

```diff
@@ -2,11 +2,23 @@
 
 #include <functional>
 
+#include "Event.h"
 #include "UIObject.h"
 
 namespace crepe {
 
-//! Represents a clickable UI button, derived from the UiObject class.
+/**
+ * \brief Button component.
+ * 
+ * This component creates a clickable surface at the transform location with the specified width and height.
+ * 
+ * The Button can be used in scripts by subscribing a EventHandler to the following events:
+ * - **ButtonPressEvent**: Triggered when the surface is clicked with the mouse. Contains GameObject metadata.
```

Comment:
Niet beschrijven wat er in de event zit, dat staat al bij de documentatie voor dat event.